### PR TITLE
mgr/dashboard: Allow the user to add the access/secret key on zone edit and not on zone creation

### PR DESCRIPTION
**Pull request by aaSharma14**
[Screencast from 2024-02-13 14-01-53.webm](https://github.com/ceph/ceph/assets/66050535/ecc6b126-e372-4fef-8535-979789a99d2f)

In current design access/secret key is compulsory while creating zone from a dashboard
so as a workflow we cannot use any replication user(dedicated user) access/secret keys during zone creation since that user will not be accessible post creating new default real/zg/zones.
due to this reason we have to use "default dashboard user" while creating zone and then need to edit the zone with any specific replication user details.

Fixes: https://tracker.ceph.com/issues/64080

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by Pegonzal — COMMENTED** [review]
Hey @aaSharma14, while testing it I found an issue when you add a zone without any access/scret key and later on you try to edit this zone (without adding an access/secret key)

edit: for the default zone

Traceback:
```
Traceback (most recent call last):
  File "/ceph/src/pybind/mgr/dashboard/services/exception.py", line 47, in dashboard_exception_handler
    return handler(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/cherrypy/_cpdispatch.py", line 54, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/ceph/src/pybind/mgr/dashboard/controllers/_base_controller.py", line 263, in inner
    ret = func(*args, **kwargs)
  File "/ceph/src/pybind/mgr/dashboard/controllers/_rest_controller.py", line 191, in wrapper
    return func(*vpath, **params)
  File "/ceph/src/pybind/mgr/dashboard/controllers/rgw.py", line 1041, in set
    })
  File "/ceph/src/pybind/mgr/dashboard/services/rgw_client.py", line 1496, in edit_zone
    try:
  File "/ceph/src/pybind/mgr/dashboard/services/rgw_client.py", line 1446, in modify_zone
    '--data-pool', data_pool,
  File "/ceph/src/pybind/mgr/dashboard/services/rgw_client.py", line 1378, in update_period
    cmd_create_zone_options.append('--secret')
dashboard.exceptions.DashboardException: 2024-01-19T08:13:46.228+0000 7fd3482bb780 -1 asok(0x556d43fd8000) AdminSocketConfigObs::init: failed: AdminSocket::bind_and_listen: failed to bind the UNIX domain socket to '/ceph/build.ceph/out/mgr.x.asok': (17) File exists
2024-01-19T08:13:46.237+0000 7fd3482bb780  0 period (ad78ceae-6649-4487-928b-59165f184807 does not have zone c126c5c8-394e-4ce3-8b10-f25ded9bb67b configured
Sending period to new master zone 35cb099d-5e32-426b-821d-deb829154d5f
request failed: (13) Permission denied
failed to commit period: (13) Permission denied
```

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by cloudbehl** [comment]
@aaSharma14 please add recording

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.html [hunk 1]
```diff
@@ -119,15 +119,18 @@
         </label>
         <div class="cd-col-form-input">
           <input class="form-control"
-                 type="text"
+                 type="password"
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.html:122]: when you change the type to password, you'll need to provide the eye icon to show the password.

Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.html:122]: You can check the references in the password field that we normally use across dashboard

Comment (aaSharma14) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.html]: Updated the PR

**Comment by aaSharma14** [comment]
jenkins test windows

**Comment by aaSharma14** [comment]
jenkins test make check

**Comment by aaSharma14** [comment]
jenkins test make check

**Review by cloudbehl — APPROVED** [review]

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.html [hunk 2]
```diff
@@ -110,37 +110,55 @@
...
+                   id="access_key"
+                   name="access_key"
+                   formControlName="access_key">
+            <button type="button"
+                    class="btn btn-light"
+                    cdPasswordButton="access_key">
+            </button>
+            <span class="invalid-feedback"
+                  *ngIf="multisiteZoneForm.showError('access_key', formDir, 'access_required')"
+                  i18n>This field is required.</span>
```
Comment (Pegonzal) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.html:134]: This is not needed, right? Is no longer a required field

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.html [hunk 3]
```diff
@@ -110,37 +110,55 @@
...
+                   id="secret_key"
+                   name="secret_key"
+                   formControlName="secret_key">
+            <button type="button"
+                    class="btn btn-light"
+                    cdPasswordButton="secret_key">
+            </button>
+            <span class="invalid-feedback"
+                  *ngIf="multisiteZoneForm.showError('secret_key', formDir, 'access_required')"
+                  i18n>This field is required.</span>
```
Comment (Pegonzal) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-zone-form/rgw-multisite-zone-form.component.html:160]: Same thing here

**Review by Pegonzal — COMMENTED** [review]
Looks good @aaSharma14, just a couple things I saw:

**Review by Pegonzal — APPROVED** [review]
Thanks @aaSharma14 ! I'll merge it once the test are passing

**Comment by Pegonzal** [comment]
jenkins test make check

**Pull request merged**